itests/eth_filter_test.go

```diff
@@ -137,6 +137,40 @@ func TestEthNewPendingTransactionFilter(t *testing.T) {
 	}
 }
 
+func TestEthNewHeadsSub(t *testing.T) {
```

Comment:
can you document inside the test here, maybe near the assertions in the listener, how it relates to the fix? it doesn't seem to be testing much other than the ability to look up block numbers as they come in. Is that because it'd be too hard (racy) to test the relationship between the block we've received and the current head so we can't really do a "are we behind by one" properly?